theseus_gui/src/main.js

```diff
@@ -11,13 +11,57 @@ import loadCssMixin from './mixins/macCssFix.js'
 import { get } from '@/helpers/settings'
 import { invoke } from '@tauri-apps/api'
 import { isDev } from './helpers/utils.js'
+import { createPlugin } from '@vintl/vintl/plugin'
+import { localeDefinitions } from '@modrinth/theseus/locales/index.js'
+
+const vintl = createPlugin({
+  controllerOpts: {
+    // From our locale definitions map every key to its own locale.
+    locales: Object.keys(localeDefinitions).map((tag) => ({ tag })),
+    // Pass our event listeners
+    listen: {
+      // ... one of which is listener for locale loading.
+      async localeload(e) {
+        // Read the tag of the locale that's currently loading.
+        const { tag: locale } = e.locale
+
+        // It's not an issue if we fail to load the locale, so wrap this in a try-catch block.
+        try {
+          // Get our definition.
+          const definition = localeDefinitions[locale]
+          // Sanity check that we indeed have this locale's definition.
+          //
+          // Because it would be a shame if some person called "Brawaru" tried to like, use
+          // console to add a custom locale for testing purposes, and that locale never actually
+          // existed in Omorphia. A big big shame.
```

Comment:
You didn't really need to copy all of these comments from the example 😂 They were just to describe the train of thought that went into writing the example, in case someone else may spot any mistakes or come up with a better idea how some things can be implemented. The code should still be pretty clear without most of them 🙂

Reply:
tbh i was just happy that it compiled again

removed most(probably to many) comments